lib/std/Thread.zig

```diff
@@ -76,6 +76,188 @@ pub fn spinLoopHint() void {
     }
 }
 
+const maxNameLen = switch (std.Target.current.os.tag) {
+    .linux => 15,
+    .windows => 31,
+    .macos, .ios, .watchos, .tvos => 63,
+    .netbsd => 31,
+    .freebsd => 15,
+    .openbsd => 31,
+    else => 0,
+};
+
+pub fn setName(self: Thread, name: []const u8) !void {
+    if (name.len > maxNameLen) return error.NameTooLong;
+
+    const name_with_terminator = blk: {
+        var name_buf: [maxNameLen:0]u8 = undefined;
+        std.mem.copy(u8, &name_buf, name);
+        name_buf[name.len] = 0;
+        break :blk name_buf[0..name.len :0];
+    };
+
+    switch (std.Target.current.os.tag) {
+        .linux => if (use_pthreads) {
+            const err = c.pthread_setname_np(self.data.handle, name_with_terminator.ptr);
+            return switch (err) {
+                0 => {},
+                os.ERANGE => unreachable,
+                else => return os.unexpectedErrno(err),
+            };
+        } else if (self.data.handle == getCurrentId()) {
+            const err = try os.prctl(.SET_NAME, .{@ptrToInt(name_with_terminator.ptr)});
+            return switch (err) {
+                0 => {},
+                else => return os.unexpectedErrno(err),
+            };
+        } else {
+            var buf: [32]u8 = undefined;
+            const path = try std.fmt.bufPrint(&buf, "/proc/self/task/{d}/comm", .{self.data.handle});
+
+            const file = try std.fs.cwd().openFile(path, .{ .write = true });
+            defer file.close();
+
+            try file.writer().writeAll(name);
+        },
+        .windows => if (std.Target.current.os.isAtLeast(.windows, .win10_rs1)) |res| {
+            // SetThreadDescription is only available since version 1607, which is 10.0.14393.795
+            // See https://en.wikipedia.org/wiki/Microsoft_Windows_SDK
+            if (!res) {
+                return error.Unsupported;
+            }
+
+            var name_buf_w: [maxNameLen:0]u16 = undefined;
+            const length = try std.unicode.utf8ToUtf16Le(&name_buf_w, name);
+            name_buf_w[length] = 0;
+
+            try os.windows.SetThreadDescription(
+                self.data.handle,
+                std.meta.cast(os.windows.LPWSTR, name_buf_w[0..length :0]),
```

Comment:
Nit, `meta.cast` is not supposed to be used outside of `translate-c`'d code. A `@ptrCast` is enough here.

Reply:
Ok, I never realized that.

Reply:
Imho, a good ol `&name_buf_w` should do the job here, no need to slice (we know it's NUL terminated anyways) and the function seems to accept a `[*:0]const u16`.